sdks/client-go/README.md

```diff
@@ -146,16 +146,23 @@ Some rollout strategies need to be calculated per-request, which means that we c
 		Platform: "linux",
 		Country:  "New Zealand",
 		Version:  "1.0.5",
+		Custom: map[string]interface{}{
+			"startDate": "now",
+			"username":  "prawn",
+			"iteration", float64(5),
+		},
 	}
 
-	// First retrieve the value as a feature:
-	featureValue, _ := fhClient.GetFeature("feature-key")
-
-	// Then you can apply a client context and read your value as its specific type, with rollout strategies applied:
-	numberValue, _ := featureValue.WithContext(clientContext).AsNumber()
+	// Use client.WithContext() to retrieve a string-value with client-side evaluation:
```

Comment:
This is a breaking change to the API, but it means that you can get an evaluated feature as a one-liner. Worth it in the long run I think.

Reply:
we have in the other APIs gone to doing this the other way around, so it would be:

in typescript this is clientContext.feature('FEATURE_NAME').asNumber()
in c# it is clientContext['FEATURE_NAME'].number
in Java it is clientContext.feature("FEATURE_NAME").getNumber()

so we are trying to be context first, the user creates a config (with the host name and key as you have done), and then asks that config for a new context. By default it will create a new repository and a new edge client if not provided one, but it can provide a repository and provide a replacement edge client (you have two, the SSE and polling one).

sort of like:
config = new EdgeProvider("http://localhost:8085", "default/1234*564")
context (Irina prefers client) = config.NewContext();
context.feature(....)

My recommendation is if you see an API key with no * in it, it means its a server side key and you should fail the request. All client side keys have an * in them. 

I then just followed your pattern with the .WithContext - so I store the link to the repository in the context, and then just go repository.feature(X).WithContext(this).AsNumber()